### DISCUSSION TITLE
配置成功，一肚子话跟大家分享、讨论

### DISCUSSION THREAD

**Discussion by ruliuyan** [vernesong/OpenClash#3955] [General]
23年早期吧，第一次因为工作找有代理功能的路由器，最开始以为随便买一个就行，后来发现不是那么一回事……
以前一直用的PC和移动端的v2ray，但是这次的需求是要控制若干台mac设备，并且每台设备固定绑定一个节点，默认还有一个平时出海查资料的节点。兜兜转转找到openwrt，一年多时间来来回回至少编译了100多个版本，从最开始的接触到后面实属是玩闹的性质。
最开始用的ssrplus，配置好后发现不满足需求，又找了一圈发现passwall可以，并且实践了也一直在用，同时知道了openclash肯定也可以，但是从各种渠道了解到配置很复杂，当时就没折腾了。
这几天又想起这事，又编译了一个带openclash的openwrt在虚拟机上折腾。以前其实有过两次实质性接触，第一次进ui的时候都蒙了！无数的tab选项卡扑面而来，菜单挨个儿点了一下。嗯，没看懂……
光是规则配置页面至少5/6个！从一个技术一个开发人员来讲，我第一时间思考的是这玩意儿没产品经理，或者产品被开发打死了。就像dbviewer一样，整个界面充斥着一股技术的味道，甭管好不好看，布局合不合理，反正功能就直接怼在你脸上，你就说功能多不多强不强大吧！
第二时间思考的是这么多规则配置页面，哪些是互不相关的、哪些是互相补充的、哪些是互相排斥的、哪些是互相覆盖的，覆盖的话优先级又是怎样的呢？
那第一次是因为什么最后放弃研究了呢？找到内核页面，嗯，看到3种内核dev/tun/meta，嗯……直接蒙了好吧！找资料了解3种有什么不同……看完后就弃了……当时时间实在不够，passwall也满足需求。
第二次呢，又有点时间，又想折腾一下。嗯，倒在了模板那，提供了一堆模板，正想开开心心用，结果呢，可能有后门！不是，提供一堆模板又告诉我可能有后门，那不提供了个寂寞，是为了让我去学习配置语法吗……从咱需求就能看出来，咱不用机场……所以……一切都得手动配置！查了些资料，当时就麻了！合着我得从盘古开天辟地开始配置？不是，openclash不是个产品吗？我怎么感觉它只是个内核……放弃！
第三次，就是今天！为什么孜孜不倦一定要搞！就是为了争口气，我要看那个监控界面，看一眼就行！看了wiki，扒拉了几个从入门到入土的视频。好吧，多数视频都是机场主的，完全是“你就这样一键配置就行了”的废话……经过“漫长”时间的捣鼓，终于……看到了监控页面……嗯，心满意足了，关掉虚拟机、释放虚拟机……真的！如果说ssrplus/psw/hw/bp等是产品，openclash最多算是个dos窗口，不能再多了！
就像这样：
对于ssrplus/psw/hw/bp
开发：你觉得这产品好用吗
用户：很好用
开发：不，我觉得它还不够傻瓜！
用户：优秀(大拇指)

对于openclash
开发：你觉得这产品好用吗
用户：好用？你是不是对好用有什么误解！来，大声告诉我！我TM爬个楼梯都恐高，你这学习曲线是直接让我蹦极？！！！
开发：哦……默默的掏出了键盘递了过去……想要什么功能：自！己！敲！（一脸认真）
用户：@#￥@#@%#&%……*&……#擦！

总结一下：openclash和其它同类产品走的就不是一个路线！其它产品是把用户当傻子，恨不得填个节点、用户、密码就完事；但是openclash走了另外一条路，它不能算是一个产品，更像是一个是内核、一个调制器、一个IDE，要什么功能自己去学自己去编！从产品角度openclash是不及格的，从功能角度它又足够强能让有经验的用户享受到完全控制的乐趣！

**Comment by markixy** [comment]
我正在经历这种过程，起因就是我听到朋友说路由器刷openwrt可以科学上网。我觉得很酷，开搞。

**Comment by refined-fish** [comment]
哈哈哈，还真有一点，使用openclash本质上是在使用clash内核，也就是看中clash强大的分流，用了好几年。
现在openclash对我来说几乎已经变成了一个启动器，所有的配置文件都被我用自己编写的模板转换进了订阅里面，openclash中所有的设置项几乎都没使用。

**Reply by mqcycy** [reply]
我摸索完，从手搓配置文件到搭建订阅转换，最后发现openclash、mihomo正确的食用方法就是配置订阅，给它一个远程配置订阅地址，然后所有的操作都可以通过编辑配置文件完成，本地点个重载即可，

**Reply by refined-fish** [reply]
太对了🤗，现在有些可惜的就是mihomo不支持设置dns为
```- dhcp://@wan```
可以设置的话就能自适应不同机器的wan口dhcp的dns（也就是给cn域名配置上运营商dns达到最快解析）

**Reply by mqcycy** [reply]
支持 - system

**Reply by refined-fish** [reply]
> 支持 - system

那个的意思是使用53端口，查询的是DNSmasq，不是接口DHCP的服务器

**Comment by screousi0103** [comment]
更复杂了

其实openclash就应该默认只提供一个页面

只能看见四个分区
- 运行状态
- 控制面板跳转
- 导入配置或订阅链接
- 启动停止

其余选项和页面全部做隐藏处理
省的新手上来一看到处都是选项

能启动成功在考虑修改配置文件

**Reply by refined-fish** [reply]
你需要的是mihomoTproxy，我也觉得老手这个插件更好用
